resources/splunk_service.rb

```diff
@@ -22,12 +21,18 @@ class SplunkService < Chef::Resource
 
   def after_created
     package_from_name unless property_is_set?(:package) || property_is_set?(:install_dir)
+    delayed_action :__guarded_restart if run_context.parent_run_context.nil?
```

Comment:
What case is this meant to cover?  A comment with some explanation of why we're checking the parent run context would help here.